tests/unallowed_relayer.go

```diff
@@ -0,0 +1,90 @@
+package tests
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
+	"github.com/ava-labs/teleporter/tests/network"
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+func UnallowedRelayerGinkgo() {
+	UnallowedRelayer(&network.LocalNetwork{})
+}
+
+func UnallowedRelayer(network network.Network) {
+	var (
+		teleporterMessageID *big.Int
+	)
+
+	subnets := network.GetSubnetsInfo()
+	subnetAInfo := subnets[0]
+	subnetBInfo := subnets[1]
+	teleporterContractAddress := network.GetTeleporterContractAddress()
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	subnetATeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetAInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+	subnetBTeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetBInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+
+	//
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to Subnet B
+	// The Teleporter message includes an allowed relayer list that does NOT include the relayer
+	//
+	ctx := context.Background()
+
+	sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: subnetBInfo.BlockchainID,
+		DestinationAddress: fundedAddress,
+		FeeInfo: teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: fundedAddress,
+			Amount:          big.NewInt(0),
+		},
+		RequiredGasLimit: big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{
+			common.HexToAddress("0x0123456789012345678901234567890123456789"),
+		},
+		Message: []byte{1, 2, 3, 4},
+	}
+	signedTx := utils.CreateSendCrossChainMessageTransaction(
+		ctx, subnetAInfo, sendCrossChainMessageInput, fundedAddress, fundedKey, teleporterContractAddress,
+	)
+
+	log.Info(
+		"Sending Teleporter transaction on source chain",
+		"destinationChainID", subnetBInfo.BlockchainID,
+		"txHash", signedTx.Hash(),
+	)
+	receipt := utils.SendTransactionAndWaitForAcceptance(ctx, subnetAInfo, signedTx, true)
+
+	event, err := utils.GetEventFromLogs(receipt.Logs, subnetATeleporterMessenger.ParseSendCrossChainMessage)
+	Expect(err).Should(BeNil())
+	Expect(event.DestinationChainID[:]).Should(Equal(subnetBInfo.BlockchainID[:]))
+
+	teleporterMessageID = event.Message.MessageID
+
+	//
+	// Relay the message to the destination
+	//
+
+	network.RelayMessage(ctx, receipt, subnetAInfo, subnetBInfo, false, false)
```

Comment:
can we also check that relayer with the allowed relayer is successful?

Reply:
IMO that's sufficiently covered in other test cases. Adding that here would require modifying the relayer interface to accept a private key, which I don't think is worth it for an unnecessary change.